### DISCUSSION TITLE
[A-L] (27 marzo 2024) Esercizio 4

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/progettazione-di-algoritmi#28] [Esercitazioni A-L]
Come cambia il numero di componenti fortemente connesse di un grafo diretto se viene aggiunto un nuovo arco?

**Comment by Frisbon** [comment]
Dalla sacra Bibbia sappiamo che:
> Due componenti fortemente connesse non hanno nodi in comune e sono massimali.

Supponiamo $H$ e $S$ due componenti.
![image](https://github.com/sapienzastudentsnetwork/progettazione-di-algoritmi/assets/84731341/135096e0-34c1-455c-b0bb-60f7d7396cd0)

Se aggiungo un arco tra due nodi di $H$, il numero di componenti rimane identico.
![image](https://github.com/sapienzastudentsnetwork/progettazione-di-algoritmi/assets/84731341/183b11e6-02cb-435d-aee1-0c15c031a4d3)

Se aggiungo un arco tra un nodo di $H$ ed un nodo di $S$ possono verificarsi due scenari:
- Esiste un cammino da $H$ ad $S$ e viceversa (ciclo), in tal caso la componente diventa unica $(H\cup S)$ e quindi il numero scende di uno.
![image](https://github.com/sapienzastudentsnetwork/progettazione-di-algoritmi/assets/84731341/8ba4b371-106e-4624-b9b9-f38d9fd092e5)
![image](https://github.com/sapienzastudentsnetwork/progettazione-di-algoritmi/assets/84731341/fc155085-93b6-4f24-befb-b62f82417884)

- Altrimenti il numero non viene modificato.

![image](https://github.com/sapienzastudentsnetwork/progettazione-di-algoritmi/assets/84731341/cd0bb511-415a-4ef8-9696-004913d8945f)
